test/functional/sqlsrv/sqlsrv_ae_fetch_phptypes.phpt

```diff
@@ -0,0 +1,205 @@
+{
+    // Each value must be inserted twice because the AE and non-AE column are side by side.
+    $testValues = array();
+    for ($i=0; $i<sizeof($values[$v]); ++$i) {
+        $testValues[] = $values[$v][$i];
+        $testValues[] = $values[$v][$i];
+    }
+
+    // Insert the data using sqlsrv_prepare()
+    // Insert one set of data for each PHPTYPE
+    $stmt = sqlsrv_prepare($conn, $insertQuery, $testValues);
+    if ($stmt == false) {
+        print_r(sqlsrv_errors());
+        fatalError("sqlsrv_prepare failed\n");
+    }
+
+    for ($i = 0; $i < sizeof($SQLSRV_PHPTYPE_CONST); ++$i) {
+        if (sqlsrv_execute($stmt) == false) {
+            print_r(sqlsrv_errors());
+            fatalError("sqlsrv_execute failed\n");
+        }
+    }
+
+    $selectQuery = "SELECT * FROM $tableName";
+
+    // Two select statements for selection using 
+    // sqlsrv_get_field and sqlsrv_fetch_array
+    $stmt = sqlsrv_query($conn, $selectQuery);
+    if ($stmt == false) {
+        print_r(sqlsrv_errors());
+        fatalError("First sqlsrv_prepare failed\n");
+    }
+    
+    $stmt2 = sqlsrv_query($conn, $selectQuery);
+    if ($stmt2 == false) {
+        print_r(sqlsrv_errors());
+        fatalError("Second sqlsrv_prepare failed\n");
+    }
+
+    $metadata = sqlsrv_field_metadata($stmt);
+    $numFields = count($metadata);
+
+    $i = 0;
+    $valueAE = null;
+    $valueArrayAE = null;
+    
+    while ($result = sqlsrv_fetch($stmt)) {
+        $dataArray = sqlsrv_fetch_array($stmt2, SQLSRV_FETCH_NUMERIC);
+        
+        for ($j = 0; $j < $numFields; $j++) {
+            $value = sqlsrv_get_field($stmt, $j, $SQLSRV_PHPTYPE_CONST[$i]);
+            $valueArray = $dataArray[$j];
+            
+            // PHPTYPE_STREAM returns a PHP resource, so check the type
+            if (is_resource($value)) $value = get_resource_type($value);
+
+            // For each type, the AE values come first and non-AE values second
+            // So let's do the comparison every second field
+            if ($j%2 == 0) {
+                $valueAE = $value;
+                $valueArrayAE = $valueArray;
+            } elseif ($j%2 == 1) {
+                // If returning a DateTime PHP type from a date only SQL type,
+                // PHP adds the current timestamp to make a DateTime object,
+                // and in this case the AE and non-AE times may be off by a 
+                // fraction of a second since they are retrieved at ever-so-slightly
+                // different times. This not a test-failing discrepancy, so 
+                // below the DateTime objects are made equal again for the next if
+                // block.
+                if ($value instanceof DateTime) {
+                    // date_diff returns a DateInterval object, and f is
+                    // the difference expressed as a fraction of a second
+                    $datediff = date_diff($value, $valueAE);
+                    $diff = $datediff->s; 
```

Comment:
should explain when the number of seconds of a DateInterval is retrieved, it should be 0 since the time difference between fetching the AE and non-AE column is likely to be less than a second